main/test/ca/uwaterloo/flix/library/TestMutList.flix

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright 2017 Esben Bjerre
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace TestMutList {
+
+/////////////////////////////////////////////////////////////////////////////
+// length                                                                  //
+/////////////////////////////////////////////////////////////////////////////
+
```

Comment:
I did not check all of these myself, but my assumption is that you have 3-5 test cases for each public function.

For critical functions ("where something funny happens") e.g. `insert` you may want to have a few more to ensure that everything is working as expected.